koru-compiler-plugin/src/main/kotlin/com/futuremind/koru/gradle/CompilerPlugin.kt

```diff
@@ -0,0 +1,66 @@
+package com.futuremind.koru.gradle
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.create
+import org.gradle.kotlin.dsl.dependencies
+import org.gradle.kotlin.dsl.getByType
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+
+
+class CompilerPlugin : Plugin<Project> {
+
+    override fun apply(project: Project) = with(project) {
+        val extension: KoruPluginExtension = extensions.create("koru")
+        requireKspPluginDependency()
+        enableKspRunForCommonMainSourceSet()
+        makeSureCompilationIsRunAfterKsp()
+        afterEvaluate {
+            requireSourceSetsNamesSet(extension.nativeSourceSetNames)
+            addGeneratedFilesToSourceSets(extension.nativeSourceSetNames)
+        }
+    }
+
+    private fun Project.requireKspPluginDependency(){
+        requireNotNull(pluginManager.findPlugin("com.google.devtools.ksp")) {
+            "You need to provide ksp plugin, e.g. plugins { id(\"com.google.devtools.ksp\") version \"1.7.0-1.0.6\" }"
+        }
+    }
+
+    private fun Project.enableKspRunForCommonMainSourceSet() = dependencies {
+        //todo don't hardcode version
```

Comment:
This probably requires manual parsing of gradle.properties, need to take a look. Not a blocker for merge, though